src/Application.Contracts/PlexAccounts/CheckUsername/CheckUsernameTaskRequest.cs

```diff
@@ -0,0 +1,27 @@
+using FluentResults;
+using MediatR;
+
+namespace Application.Contracts;
+
+public class CheckUsernameTaskRequest : IRequest<Result<bool>>
```

Comment:
As mentioned previously, consider using a `record` instead of a `class` .
You would turn a 21 lines class into a single line.
Regarding naming, it could be renamed as `CheckUsernameQuery`, given it's a Query

Reply:
I agree with using a `record` instead of a `class ` and I was in conflict with the naming as well. 

So now I have two "types" of query/handlers
1. The query mediatr handlers that do queries on the database and return the data
2. The query/task in this case that contains business logic and non-database related code

I figured I could separate these by name by having (1) end with `query` and (2) end with `task`. What do you think?

Reply:
1. Yep, a `Query` returns data and does not produce a side effect
2. You should not use `Query` or `Task` here; use `Command` instead. A command produces a side-effect and does not return any data
This is what CQRS (Command-Query Responibilty Segregation) or CQS (Command-Query Separation) recommends